### PR TITLE
Syslog rate limit design

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**

### CODE REVIEW

**Review by stephenxs — APPROVED** [review]
LGTM.

File: doc/syslog/syslog-rate-limit-design.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+Currently, each container has hardcoded message rate limiting to avoid receiving flooded log messages:
+
+```
+$SystemLogRateLimitInterval 300
+$SystemLogRateLimitBurst 20000
+```
+
+The SystemLogRateLimitInterval determines the amount of time that is being measured for rate limiting. The SystemLogRateLimitBurst defines the amount of messages, that have to occur in the time limit of SystemLogRateLimitInterval, to trigger rate limiting.
+
+This feature allows user to configure SystemLogRateLimitInterval and SystemLogRateLimitBurst for host and containers.
```
Comment (Junchao-Mellanox) [inline doc/syslog/syslog-rate-limit-design.md]: Should we limit syslog rate on host side?

Comment (Junchao-Mellanox) [reply doc/syslog/syslog-rate-limit-design.md]: Do a test for host side rate limit

File: doc/syslog/syslog-rate-limit-design.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+                type string {
+                    pattern "host";
+                }
+            }
+        }
+
+        leaf rate_limit_interval {
+            description "Message rate limit interval";
+            type uint32 {
+                range 1..100000;
```
Comment (Junchao-Mellanox) [inline doc/syslog/syslog-rate-limit-design.md]: consider removing the range...or, check the range from rsyslog official website.

File: doc/syslog/syslog-rate-limit-design.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            }
+        }
+
+        leaf rate_limit_burst {
+            description "Message rate limit burst";
+            type uint32 {
+                range 1..1000000;
```
Comment (Junchao-Mellanox) [inline doc/syslog/syslog-rate-limit-design.md]: consider removing the range...or, check the range from rsyslog official website.

File: doc/syslog/syslog-rate-limit-design.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+### Warmboot and Fastboot Design Impact
+
+N/A
+
+### Restrictions/Limitations
+
+1. Cannot support container not registered to FEATURE table
+2. Persist syslog configuration for database container will not be loaded on next reboot/reload. (Open question)
+3. To support this feature in APP extension, APP extension need meet below requirements:
```
Comment (Junchao-Mellanox) [inline doc/syslog/syslog-rate-limit-design.md]: talk with Stepan about app extension.

**Pull request closed without merging**